test/deflateparams.c

```diff
@@ -0,0 +1,261 @@
+    }
+
+    err = PREFIX(deflateEnd)(&c_stream);
+    CHECK_ERR(err, "deflateEnd");
+}
+
+
+/* ===========================================================================
+ * deflate() with large buffers
+ */
+void deflate_params_large(unsigned char *buf, uint32_t len, int level, int window_bits, int mem_level, int strategy)
+{
+    PREFIX3(stream) c_stream; /* compression stream */
+    unsigned char compr[4096];
+    int err;
+
+    c_stream.zalloc = zalloc;
+    c_stream.zfree = zfree;
+    c_stream.opaque = (void *)0;
+    c_stream.total_in = 0;
+    c_stream.total_out = 0;
+
+    err = PREFIX(deflateInit2)(&c_stream, level, Z_DEFLATED, window_bits, mem_level, strategy);
+    CHECK_ERR(err, "deflateInit2");
+    
+    c_stream.next_in  = (const unsigned char *)buf;
+    c_stream.next_out = (unsigned char *)compr;
+
+    while (c_stream.total_in < len) {
+        c_stream.avail_in = c_stream.avail_out = sizeof(compr); /* Force large buffers */
+        if (c_stream.avail_in > (len - c_stream.total_in))
+            c_stream.avail_in = (len - c_stream.total_in);
+        err = PREFIX(deflate)(&c_stream, Z_NO_FLUSH);
+        CHECK_ERR(err, "deflate");
+        fwrite(compr, 1, c_stream.next_out - compr, stdout);
+    }
+
+    /* Finish the stream using large buffers */
+    c_stream.avail_in = 0;
+    do {
+        c_stream.avail_out = sizeof(compr);
+        err = PREFIX(deflate)(&c_stream, Z_FINISH);
+        fwrite(compr, 1, c_stream.next_out - compr, stdout);
+        if (err == Z_STREAM_END) break;
+        CHECK_ERR(err, "deflate");
+    } while (err == Z_OK);
+
+    err = PREFIX(deflateEnd)(&c_stream);
+    CHECK_ERR(err, "deflateEnd");
+}
+
+/* ===========================================================================
+ * deflate() with params
+ */
+void deflate_params(unsigned char *buf, uint32_t len, int level, int window_bits, int mem_level, int strategy)
+{
+    PREFIX3(stream) c_stream; /* compression stream */
+    size_t compr_bound;
+    unsigned char *compr;
+    int err;
+
+    c_stream.zalloc = zalloc;
+    c_stream.zfree = zfree;
+    c_stream.opaque = (void *)0;
+    c_stream.total_in = 0;
+    c_stream.total_out = 0;
+
+    compr_bound = PREFIX(compressBound)(len);
+    compr = malloc(compr_bound);
+    assert(compr != NULL);
+
+    err = PREFIX(deflateInit2)(&c_stream, level, Z_DEFLATED, window_bits, mem_level, strategy);
+    CHECK_ERR(err, "deflateInit2");
+    
+    c_stream.next_in  = (const unsigned char *)buf;
+    c_stream.avail_in = len;
+    c_stream.next_out = compr;
+    c_stream.avail_out = (uint32_t)compr_bound;
+
+    err = PREFIX(deflate)(&c_stream, Z_FINISH);
+    if (err != Z_STREAM_END)
+        CHECK_ERR(err, "deflate");
+
+    err = PREFIX(deflateEnd)(&c_stream);
+    CHECK_ERR(err, "deflateEnd");
+
+    fwrite(compr, 1, c_stream.total_out, stdout);
+    free(compr);
+}
+
+/* ===========================================================================
```

Comment:
This is the help text. Although it would have been nice if it could print that instead of only being a comment in the source.

Reply:
I think maybe you selected the wrong lines but I think I know what you are talking about.

Reply:
I think maybe you selected the wrong lines but I think I know what you are talking about.

Reply:
Weird, I didn't select any lines, I put the comment on the first line of the in-code-comment. If you go see it in GitHubs code viewer, it'll be pretty obvious.
I guess GitHub expects all comments to come after the code.

Reply:
I agree, this should be printed on console if amount of command-line parameters is wrong or user specifies "--help" explicitly...

Reply:
I have noticed that switchlevels and minigzip are also missing help information. Maybe we can do all three apps at once in a separate PR.

Reply:
Help text for minigzip and switchlevels is now in #623.

Reply:
Added help text.